### ISSUE TITLE
Contributing back changes

### ISSUE THREAD

**Issue by Sighery** [thoughtfault/gonjalla-DeimosC2#1]
Hey, do you want to contribute back your changes? The main reason why the `gonjalla` library and `njalla` providers only support DNS records is that this all started (and still is, I'm not affiliated in any way, nor do I look for sponsorships/donations) as a way for me to learn some Golang, Terraform, and managing my DNS records on my Njalla domains.

I didn't implement other stuff (although recently someone added some methods to `gonjalla` to query domains), because I just didn't need it and wasn't even sure other people would find it useful either, so I didn't bother spending the effort.

But if the library/provider are used, and server functionality is needed in the wild, it'd be nice to have it in the main repos so it doesn't require people to manually compile and deploy custom library versions.

If you're not interested in contributing back your changes, could I use your code, either copying it outright, or basing my work off it, to add it to the library/provider? Was thinking of getting back into Golang again lately and maybe trying to refactor the DNS stuff using generics, so I'd probably use the chance to add this functionality as well.

If you're interested in contributing and potentially maintaining it as well, we can look into making you a contributor. Never had an open source project used by anyone other than me so honestly not quite sure how it works, but should be possible to set up so you can work directly off the main repo instead of forks, and so maintain/review PRs as well.

**Comment by thoughtfault** [comment]
Yes, I will contribute them back.  I would also be open to maintaining as well.
